File: Solid Principles/I - Interface Segregation Principle/Example 1/correct_example.py
class WaiterOperation:
    def serve_customer(self):
        pass
    def collect_bill(self):
        pass

class Waiter(WaiterOperation):
    def serve_customer(self):
        print('Serve Customer')
    
    def collect_bill(self):
        print('collect bill')

class ChefOperation:
    def collect_kitchen_ticket(self):
        pass
    def collect_food(self):
        pass

class Chef(ChefOperation):
    def collect_kitchen_ticket(self):
        print('collect kitchen ticket') 

    def collect_food(self):
        print('collect food')

File: Solid Principles/I - Interface Segregation Principle/Example 0/CorrectExample.java
interface WaiterOperations{
    void serveCustomer();
    void collectBill();
}

interface ChefOperations{
    void collectKitchenTicket();
    void cookFood();
}

interface CashierOperations{
    void billOrder();
    void returnChange();
}

class Waiter implements WaiterOperations{

    @Override
    public void serveCustomer() {
       // serve customer
    }

    @Override
    public void collectBill() {
        // collect Bill
    }
}

class Chef implements ChefOperations{
    @Override
    public void collectKitchenTicket() {
        // Collection KOT
    }

    @Override
    public void cookFood() {
        // cook Food
    }
}



public class CorrectExample {
   public static void main(String[] args) {
    
   }    
}
